cpp/include/cudf/strings/string.cuh

```diff
@@ -54,6 +54,13 @@ inline __device__ bool is_integer(string_view const& d_str)
            thrust::seq, begin, end, [] __device__(auto chr) { return chr >= '0' && chr <= '9'; });
 }
 
+// @cond
+inline constexpr bool is_nan_str(const char* ptr, size_type size)
```

Comment:
This should add `__device__` qualifier, as discussed in https://github.com/rapidsai/cudf/issues/7795.